### ISSUE TITLE
remove ehcache as a dep

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#11205]
Our old hibernate config specified to use ehcache as a 2nd leve cache.  We should remove it as it is not needed.  

Code taken from Hibernate 3.3

http://grepcode.com/file/repo1.maven.org/maven2/org.hibernate/hibernate-core/3.3.0.GA/org/hibernate/cache/NoCacheProvider.java

**Comment by wezell** [comment]
So this changes our hibernate to use an internal NoCache provider as a second level cache, which is ok because we don't care about hibernates session cache and in fact it has caused us untold issues:

```
INFO  2017-04-02 17:54:16,613 com.dotcms.repackage.net.sf.hibernate.cfg.Configuration:secondPassCompile(650): processing one-to-one association property references
INFO  2017-04-02 17:54:16,613 com.dotcms.repackage.net.sf.hibernate.cfg.Configuration:secondPassCompile(675): processing foreign key constraints
INFO  2017-04-02 17:54:16,630 com.dotcms.repackage.net.sf.hibernate.dialect.Dialect:<init>(86): Using dialect: com.dotcms.repackage.net.sf.hibernate.dialect.MySQLDialect
INFO  2017-04-02 17:54:16,637 com.dotcms.repackage.net.sf.hibernate.cfg.SettingsFactory:buildSettings(70): Maximim outer join fetch depth: 2
INFO  2017-04-02 17:54:16,637 com.dotcms.repackage.net.sf.hibernate.cfg.SettingsFactory:buildSettings(74): Use outer join fetching: true
INFO  2017-04-02 17:54:16,639 com.dotcms.repackage.net.sf.hibernate.connection.ConnectionProviderFactory:newConnectionProvider(53): Initializing connection provider: com.liferay.util.dao.hibernate.DSConnectionProvider
INFO  2017-04-02 17:54:16,644 com.dotcms.repackage.net.sf.hibernate.transaction.TransactionManagerLookupFactory:getTransactionManagerLookup(33): No TransactionManagerLookup configured (in JTA environment, use of process level read-write cache is not recommended)
INFO  2017-04-02 17:54:16,645 com.dotcms.repackage.net.sf.hibernate.cfg.SettingsFactory:buildSettings(114): Use scrollable result sets: true
INFO  2017-04-02 17:54:16,645 com.dotcms.repackage.net.sf.hibernate.cfg.SettingsFactory:buildSettings(117): Use JDBC3 getGeneratedKeys(): true
INFO  2017-04-02 17:54:16,645 com.dotcms.repackage.net.sf.hibernate.cfg.SettingsFactory:buildSettings(120): Optimize cache for minimal puts: false
INFO  2017-04-02 17:54:16,645 com.dotcms.repackage.net.sf.hibernate.cfg.SettingsFactory:buildSettings(129): Query language substitutions: {}
INFO  2017-04-02 17:54:16,645 com.dotcms.repackage.net.sf.hibernate.cfg.SettingsFactory:buildSettings(140): cache provider: com.dotmarketing.db.NoCacheProvider      <----------------------
INFO  2017-04-02 17:54:16,647 com.dotcms.repackage.net.sf.hibernate.cfg.Configuration:configureCaches(1130): instantiating and configuring caches
INFO  2017-04-02 17:54:16,666 com.dotcms.repackage.net.sf.hibernate.impl.SessionFactoryImpl:<init>(119): building session factory
INFO  2017-04-02 17:54:17,293 com.dotcms.repackage.net.sf.hibernate.impl.SessionFactoryObjectFactory:addInstance(82): Not binding factory to JNDI, no JNDI name configured
INFO  2017-04-02 17:54:17,318 com.dotmarketing.startup.StartupTasksExecutor:info(58): Not running: Task00001LoadSchema
INFO  2017-04-02 17:54:17,321 com.dotmarketing.startup.StartupTasksExecutor:info(58): Running: Task00003CreateSystemRoles
```

**Comment by jgambarios** [comment]
PR: https://github.com/dotCMS/core/pull/11207

**Comment by bryanboza** [comment]
Fixed, tested on master-4.1 // Postgres // FF
